src/utils.cairo

```diff
@@ -1,6 +1,76 @@
-use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
+use core::fmt::{Display, Formatter, Error};
 use core::num::traits::{Zero, One, BitSize};
-use core::starknet::secp256_trait::Secp256PointTrait;
+use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
+
+#[derive(Copy, Drop, Debug)]
+// pub is required here as we define Hash in utils and we need to import Hash in merkle_tree.cairo
+// for arguments to merkle_root function
+pub struct Hash {
+    value: [u32; 8]
+}
+
+#[generate_trait]
+pub impl HashImpl of HashTrait {
+    #[inline(always)]
+    fn new() -> Hash {
+        Hash { value: [0; 8] }
+    }
+}
+
+impl ValueDisplay of Display<[u32; 8]> {
+    fn fmt(self: @[u32; 8], ref f: Formatter) -> Result<(), Error> {
+        let value = *self;
+        writeln!(f, "Hash ({value})")
+    }
+}
+
+impl HashDisplay of Display<Hash> {
+    fn fmt(self: @Hash, ref f: Formatter) -> Result<(), Error> {
+        let value = *self.value;
+        writeln!(f, "Hash ({value})")
```

Comment:
This is not a hex representation of the hash.
I would convert to `u256` and use `append_formatted_to_byte_array`.